common/common.go

```diff
@@ -653,3 +654,34 @@ func GetTypeAssertError(required string, received interface{}, fieldDescription
 	}
 	return fmt.Errorf("%w from %T to %s%s", ErrTypeAssertFailure, received, required, description)
 }
+
+// Batch takes a slice type and converts it into a slice of containing slices of length batchSize, and any remainder in the final batch
+// batchSize <= 0 will return the entire input slice in one batch
+func Batch[S ~[]E, E any](blobs S, batchSize int) []S {
```

Comment:
Now that this is introduced, we can rid SplitStringSliceByLimit which BTC Markets is the only consumer of.

Reply:
Nice spot 🙏 

Fixed 7c3d7fb1

Sorry it's not a straight drop in, but couldn't tolerate the strangeness going on there with tempMap.